bbb/services.py

```diff
@@ -457,6 +463,20 @@ def _isAuthorized(self, buildername, scopes):
         else:
             return True
 
+    def _refreshAllowedBuilders(self):
+        now = arrow.now().timestamp
+        if self.allowed_builders is None or (now - self.allowed_builders_age) > 300:
+            try:
+                builders = requests.get(ALL_THE_THINGS_URL).json()['builders']
```

Comment:
Can you add `timeout=something` here? And maybe break it into multiple lines so you can use `raise_for_status()`?
Also what happens if the first attempt fails, and `self.allowed_builders` stays `None` for some time? Maybe set the initial value to `set()` and skip the check if it's empty?